cores/esp32/esp32-hal-gpio.c

```diff
@@ -91,6 +91,13 @@ static InterruptHandle_t __pinInterruptHandlers[SOC_GPIO_PIN_COUNT] = {0,};
 
 extern void ARDUINO_ISR_ATTR __pinMode(uint8_t pin, uint8_t mode)
 {
+#ifdef BOARD_HAS_NEOPIXEL
```

Comment:
Could we consider moving the board-specific code to the `variants` folder?

I'm worried that the more similar features we add for specific boards, the more complex the core code will become. It would be better to handle this in the board variant folder. It would also demonstrate authors of 3rd party board packages how to add such custom functionality.

To avoid duplication of code, you can still define `RGBLedWrite` somewhere in the core, but preferably in a separate source file (esp32-hal-rgb-led.c?)

Reply:
I'm a little bit confused about what should go where.
`pinMode` -> `variants/`  ... new file?
`RGBLedWrite` -> `esp32-hal-rgb-led.c` ?

Reply:
Should be solved in https://github.com/espressif/arduino-esp32/pull/6808/commits/583e0301427b6e98a87c3fa94fa100d2fc19cb7f